js/apiml.js

```diff
@@ -0,0 +1,194 @@
+const Promise = require("bluebird");
+const eureka = require('eureka-js-client').Eureka;
+const zluxUtil = require('./util');
+
+const log = zluxUtil.loggers.apiml;
+
+const MEDIATION_LAYER_EUREKA_DEFAULTS = {
+  "preferSameZone": false,
+  "requestRetryDelay": 10000,
+  "heartbeatInterval": 3000,
+  "registryFetchInterval": 10000,
+  "fetchRegistry": false,
+  "availabilityZones": {
+    "defaultZone": ["defaultZone"]
+  }, 
+};
+
+
+const MEDIATION_LAYER_INSTANCE_DEFAULTS = {
+  instanceId: "localhost:zowe-zlux:8543",
+  app: "zlux",
+  hostName: "localhost",
+  ipAddr: "127.0.0.1", 
+  vipAddress: "localhost",
+  status: "UP",
+  dataCenterInfo: {
+    '@class': 'com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo',
+    name: 'MyOwn'
+  },
+  leaseInfo: {
+    durationInSecs: 10,
+    renewalIntervalInSecs: 10
+  },
+  metadata: {
+    "routed-services.1.gateway-url": "/api/v1",
+    "routed-services.1.service-url": "/ZLUX",
+    "routed-services.2.gateway-url": "/ui/v1",
+    "routed-services.2.service-url": "/ZLUX",
+    "routed-services.3.gateway-url": "/ws/v1",
```

Comment:
Isn't it the case that for us, /api/v1, /ui/v1, and /ws/v1 really mean the same thing? What will happen if we don't have these as distinct routes? If possible, only 1 is needed.
Consider:
https://host/api/v1/ZLUX/plugins/foo/services/bar1/baz
wss://host/api/v1/ZLUX/plugins/foo/services/bar2/baz
2 different services, but they're both apis. They also follow the v1 structure of our server namespacing.
So, I think it makes a lot more sense to have only "routed-services.1" if possible.

Reply:
This is some APIML convention. A quote from Petr Plavjanik's e-mail:
> Also the prefixes /ui/v1 and /api/v1 are there to follow conventions that we have in APIML and at the end they are both mapped to the same URL in zLUX.